packages/sanity/src/core/create/components/StartInCreateBanner.tsx

```diff
@@ -0,0 +1,151 @@
+import {LaunchIcon, SparklesIcon} from '@sanity/icons'
+import {
+  // eslint-disable-next-line no-restricted-imports
+  Button,
+  Card,
+  Flex,
+  Inline,
+  Stack,
+  Text,
+  useToast,
+} from '@sanity/ui'
+import {useCallback, useState} from 'react'
+import {TextWithTone} from 'sanity'
+
+import {isDev} from '../../environment'
+import {useTranslation} from '../../i18n'
+import {useWorkspace} from '../../studio'
+import {useSanityCreateConfig} from '../context'
+import {getCreateLinkUrl} from '../createDocumentUrls'
+import {isSanityCreateExcludedType, isSanityCreateStartCompatibleDoc} from '../createUtils'
+import {createLocaleNamespace} from '../i18n'
+import {CreateLinkingDialog} from '../start-in-create/CreateLinkingDialog'
+import {type AppIdCache} from '../studio-app/appIdCache'
+import {type CompatibleStudioAppId} from '../studio-app/fetchCreateCompatibleAppId'
+import {useStudioAppIdStore} from '../studio-app/useStudioAppIdStore'
+import {type StartInCreateBannerProps} from '../types'
+import {useSanityCreateTelemetry} from '../useSanityCreateTelemetry'
+import {createUserDocumentationUrl} from './constants'
+import {DialogPortalProvider} from './DialogPortalProvider'
+import {StartInCreateDevInfoButton} from './StartInCreateDevInfoButton'
+
+export function StartInCreateBanner(props: StartInCreateBannerProps) {
+  const {document, isInitialValueLoading} = props
+  const {appIdCache, startInCreateEnabled} = useSanityCreateConfig()
+
+  const isExcludedByOption = isSanityCreateExcludedType(props.documentType)
+  const isNewPristineDoc = !document._createdAt
+  const isStartCreateCompatible = isSanityCreateStartCompatibleDoc(props.document)
+
+  if (
+    !isNewPristineDoc ||
+    !startInCreateEnabled ||
+    isExcludedByOption ||
+    !appIdCache ||
+    !isStartCreateCompatible ||
+    isInitialValueLoading
+  ) {
+    return null
+  }
```

Comment:
Smallest nitpick in the world, wonder if it makes more sense to move these out so that the component is not even rendered / called if these things condiitons are met. I'm not sure if I'm just picking on something that has the world's smallest impact on rendering but 😂

Reply:
It might also be that doing what I suggested adds unnecessary noise to an already large component 😅

Reply:
We need to run the hook to get config, so I think its nice to have all the null checks in a single place. The intermediary component is needed to able to conditionally render based when appId is resolved.

I cannot see why the code as written would be a performance issue, but keep me honest here.

Reply:
I just thought (and maybe I'm wrong), that by having these ifs within the component that the component would _still render_ or at least take time to try before realising that it doesn't need to. But again, that might be a bastardisation of what react actually does under the hood 😅

Reply:
With this early return there is very little compute being used; the inner components are never rendered.